be/tool/main.c

```diff
@@ -59,6 +78,102 @@ static const char *betool_help = ""
 "Default BE domain stob domain location is used when this parameter is absent:"
 "\n";
 
+extern void btree_dbg_print(struct m0_be_btree *tree);
+
+
+static void scan_btree(struct m0_be_domain *dom, int print_btree)
+{
+	int                     left;
+	char                    *suffix;
+	struct m0_buf            opt;
+	struct m0_be_seg        *seg;
+	struct m0_be_0type      *objtype;
+	struct m0_cob_domain     *cdom;
+	struct stob_ad_0type_rec *rec ;
+	struct m0_balloc         *m0balloc;
+
+	seg = m0_be_domain_seg0_get(dom);
+	if (seg == NULL)
+	{
+		return;
+	}
+
+	m0_tl_for(zt, &dom->bd_0types, objtype) {
+		for (left = m0_be_segobj_opt_begin(seg, objtype, &opt, &suffix);
+		left > 0 ;
+		left = m0_be_segobj_opt_next(seg, objtype, &opt, &suffix)) {
+
+			M0_LOG(M0_ALWAYS, "object b0_name = '%s' suffix = '%s'"
+						"objtype = %p seg = %p",
+						objtype->b0_name, suffix,
+						objtype, seg );
+
+			if (  strcmp(objtype->b0_name, "M0_BE:COB") == 0)
+			{
+				cdom = *(struct m0_cob_domain**)opt.b_addr;
+				M0_LOG(M0_ALWAYS, "M0_BE:COB "
+						"cd_object_index btree = %p "
+						"cd_namespace btree = %p "
+						"cd_fileattr_basic btree = %p "
+						"cd_fileattr_omg btree = %p "
+						"cd_fileattr_ea btree = %p",
+						&cdom->cd_object_index,
```

Comment:
No need to print out all these addresses:
They can be calculated from "cdom", and they are consecutive.

Reply:
its required to print btree pointers in case of track_btree case